internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -397,40 +398,25 @@ func getInterfaceName(deviceID string) (string, error) {
 
 // enableIPV6LinkLocal function to enable the IPv6 Link Local Address on the given Interface Name
 // It will return the error
-func enableIPV6LinkLocal(interfaceName string) error {
-	baseNsenterCmd := "nsenter -t 1 -m -u -n -i -- "
-	nmcliCmdStr := baseNsenterCmd + "nmcli con add type ethernet ifname " + interfaceName + " ipv6.method link-local"
-	nmcliCmd := exec.Command("/bin/sh", "-c", nmcliCmdStr)
-	if err := nmcliCmd.Run(); err != nil {
-		return err
-	}
+func enableIPV6LinkLocal(interfaceName string, ipv6Addr string) error {
+	// Ensure to set addrgenmode and toggle link state (which can result in creating
+	// the IPv6 link local address. Ignore errors here.
+	exec.Command("ip", "link", "set", interfaceName, "addrgenmode", "eui64").Run()
+	exec.Command("ip", "link", "set", interfaceName, "down").Run()
 
-	// wait for interface to get IPv6 Link local address
-	time.Sleep(3 * time.Second)
-	link, err := netlink.LinkByName(interfaceName)
+	err := exec.Command("ip", "link", "set", interfaceName, "up").Run()
 	if err != nil {
-		return err
+		return errors.Errorf("Error setting link %s up: %v", interfaceName, err)
 	}
 
-	//returns the list of IPv6 addresses on the given interface
-	address, err := netlink.AddrList(link, netlink.FAMILY_V6)
+	err = exec.Command("ip", "addr", "add", ipv6Addr+"/64", "dev", interfaceName).Run()
```

Comment:
Why do we need both an IPv6 link local and static IP address? Shouldn't we just have the static one?

Reply:
I prefer the LL address

Reply:
I don't think there is a problem.

A link local address is anyway only be usable and unique on the local network (that is, behind the interface on which they are configured).

You can configure addresses `fe80::2` on multiple interfaces just fine. Of course, whenever you use it, you must specify which interface you mean (e.g. `ping fe80::1%ens7f0`).

Reply:
@wizhaoredhat multiple DPUs is in the works and I plan to push a first PR for that. The change will be split up across multiple PRs. That's not only a Marvell specific use case. Is there a reason to target that use case this early (i.e. during this PR)?

Reply:
@thom311 I think that is fine then. Although specifying the interface would need to be added later on. I wonder if gRPC is ok to specify interfaces using that format e.g. fe80::1%ens7f0. Or will the socket needs to be bound to ens7f0

@bn222 I think this is ok to be considered now. Thomas' approach seems fine after his explanation.

Reply:
I don't think you can meaningfully use IPv6 link local addresses without an interface name. Pretty sure you will always have to specify it, and any socket API is likely to support that.